### PR TITLE
Move Proxies to .Net Standard 2.0 for Functions V2.

### PR DESCRIPTION
**Pull request by safihamid**
work in progress...

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs [hunk 1]
```diff
@@ -28,6 +30,15 @@ public FunctionInvocationMiddleware(RequestDelegate next)
             _next = next;
         }
 
+        public static bool IsHomepageDisabled
+        {
+            get
+            {
+                return string.Equals(Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebJobsDisableHomepage),
```
Comment (christopheranderson) [inline src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs]: Would prefer this to be done via IConfiguration approach so we're not locked to environment variables as the only method of setting this.

File: src/WebJobs.Script.WebHost/Proxy/ProxyFunctionExecutor.cs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                    if (routes != null)
+                    {
+                        routes.Reverse();
+                    }
+
+                    await webJobsRouter.RouteAsync(rc);
+
+                    if (rc.RouteData != null && rc.RouteData.DataTokens != null)
+                    {
+                        functionName = rc.RouteData.DataTokens["AZUREWEBJOBS_FUNCTIONNAME"].ToString();
```
Comment (christopheranderson) [inline src/WebJobs.Script.WebHost/Proxy/ProxyFunctionExecutor.cs]: nit: should avoid string literals and use a constant

**Comment by davidparker** [comment]
hi

Any idea when this might make it onto azure portal?   really need proxies to be working, but I can get by if could have some idea if this will be resolved within the next month or so at least?

many thanks

**Comment by safihamid** [comment]
Yes it should be. Waiting for some code reviews.

File: src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs [hunk 3]
```diff
@@ -37,14 +38,31 @@ public FunctionInvocationMiddleware(RequestDelegate next)
...
+
+                if (nestedProxiesCount > 0)
+                {
+                    // if Proxy, the rest of the pipleline will be processed bt Proxies in case there are response overrides and what not.
+                    SetProxyResult(context, nestedProxiesCount, result);
+                    return;
+                }
+            }
 
+            if (result != null && !context.Response.HasStarted)
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs]: We should be able to move this block to the above `if` block. `result` should also never be `null`.

**Comment by JonathanGiles** [comment]
Does this mean proxies now work in v2 functions, or are there more things to do (e.g. a new release)? Where will the availability be announced (so I don't need to keep asking)? Thanks!

**Comment by safihamid** [comment]
@JonathanGiles Yes they will work with our next release of Functions V2. We usually have a release every couple of weeks.

**Pull request merged**